lib/oli_web/live/delivery/instructor_dashboard/instructor_dashboard_live.ex

```diff
@@ -105,7 +105,16 @@ defmodule OliWeb.Delivery.InstructorDashboard.InstructorDashboardLive do
         |> Enum.reject(fn s -> s.user_role_id != 4 end)
       end)
       |> assign_new(:assessments, fn %{students: students} ->
-        Helpers.get_assessments(socket.assigns.section, students)
+        result = Helpers.get_assessments(socket.assigns.section, students)
+
+        pid = self()
+
+        Task.async(fn ->
+          result_with_metrics = Helpers.load_metrics(result, socket.assigns.section, students)
+          send(pid, {:assessments, result_with_metrics})
+        end)
+
+        result
       end)
       |> assign_new(:activities, fn -> Oli.Activities.list_activity_registrations() end)
```

Comment:
Here is the key to the  "perceived" faster loading of the views.  We asynchronously run the three metrics queries, allowing the view to load *very* quickly, and then have the details fill in when they arrive.